### PR TITLE
Fix Failing Designs

### LINKED ISSUE #206: salsa20 failing in nightly CI
### Description

Design failing in CI.

### Expected Behavior

Design passing

### Environment report

```sh
CI always runs the latest
```

### Reproduction material

[salsa20-sky130A-sky130_fd_sc_hd.zip](https://github.com/efabless/openlane2/files/12840251/salsa20-sky130A-sky130_fd_sc_hd.zip)

### Relevant log output

```sh
/home/runner/work/openlane2/openlane2/test/designs/salsa2            
           0/src/salsa20.v:692:3: Latch inferred for signal                     
           'salsa20.core.data_out_logic.lsb_block_state14' (not all             
           control paths of combinational always assign a value)                
                                                                                
                                              : ... Suggest use of              
           always_latch for intentional latches                                 
             692 |   always @*                                                  
                 |   ^~~~~~                                                     
           %Error-LATCH:                                                        
           /home/runner/work/openlane2/openlane2/test/designs/salsa2            
           0/src/salsa20.v:692:3: Latch inferred for signal                     
           'salsa20.core.data_out_logic.lsb_block_state15' (not all             
           control paths of combinational always assign a value)                
                                                                                
                                              : ... Suggest use of              
           always_latch for intentional latches                                 
             692 |   always @*                                                  
                 |   ^~~~~~                                                     
           %Error: Exiting due to 32 error(s)
```

### PR DESCRIPTION
**Pull request by donn**
* Update OpenROAD to `0a584d1`
  * Fixes issue where post-GRT resizing run-time and memory consumption got out of hand: see https://github.com/The-OpenROAD-Project/OpenROAD/issues/4121#issuecomment-1758154668 for one example
* Disabled latch linting for `salsa20`
* Enable post-GRT resizer for `aes_core`, `spm` and `aes`
* Added hidden debug options `--force-design-dir`, `--force-run-dir`
---
Resolves #206 

[ci ets]

### CODE REVIEW

File: openlane/flows/cli.py [hunk 1]
```diff
@@ -241,13 +238,32 @@ def decorator(f):
...
+                    hidden=True,
+                    default=None,
+                ),
                 o(
                     "--last-run",
                     is_flag=True,
                     default=False,
                     help="Use the last run as the run tag.",
                 ),
-                constraint=mutually_exclusive,
```
Comment (donn) [inline openlane/flows/cli.py:250]: `--last-run` and `--tag` should still be mutually exclusive

File: openlane/flows/flow.py [hunk 2]
```diff
@@ -284,6 +284,7 @@ def __init__(
         scl: Optional[str] = None,
         design_dir: Optional[str] = None,
         config_override_strings: Optional[Sequence[str]] = None,
+        force_design_dir: Optional[str] = None,
```
Comment (donn) [inline openlane/flows/flow.py:287]: Prefix with `_` to indicate internal API

File: openlane/flows/flow.py [hunk 3]
```diff
@@ -454,10 +457,18 @@ def start(
 
         :returns: ``(success, state_list)``
         """
-        if last_run:
-            if tag is not None:
-                raise FlowException("tag and last_run cannot be used simultaneously.")
+        if last_run and tag is not None:
+            raise FlowException("tag and last_run cannot be used simultaneously.")
 
+        if run_dir is not None and tag is not None:
```
Comment (donn) [inline openlane/flows/flow.py:463]: Run dir should not be part of the API either I think. It should be _force_run_dir

**Comment by kareefardi** [comment]
Confirmed working with `859b21b9cf454de89f4e02be27926c749360060c`

File: openlane/config/config.py [hunk 4]
```diff
@@ -400,13 +401,17 @@ def load(
...
+
             if design_dir is not None:
                 raise TypeError(
                     "The argument design_dir is not supported when config_in is not a dictionary."
                 )
-            config_in = os.path.abspath(config_in)
+            if _force_design_dir is not None:
+                design_dir = _force_design_dir
+            else:
+                design_dir = str(os.path.dirname(config_in))
```
Comment (donn) [inline openlane/config/config.py:413]:
```suggestion
    design_dir = _force_design_dir or ...
```

File: openlane/flows/flow.py [hunk 5]
```diff
@@ -472,12 +484,14 @@ def start(
...
+            self.run_dir = os.path.join(self.design_dir, "runs", tag)
+        elif tag is None:
             tag = datetime.datetime.now().astimezone().strftime("RUN_%Y-%m-%d_%H-%M-%S")
+            self.run_dir = os.path.join(self.design_dir, "runs", tag)
+        else:
+            self.run_dir = os.path.join(self.design_dir, "runs", tag)
 
-        # Stored until next start()
-        self.run_dir = os.path.join(self.design_dir, "runs", tag)
+        # self.run_dir Stored until next start()
```
Comment (donn) [inline openlane/flows/flow.py:494]: S should be small

**Pull request merged**